python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -198,6 +197,25 @@ def _set_vxlan_interface_config_for_vlan(
                 tenant.evpn_l2_multicast.underlay_l2_multicast_group_ipv4_pool_offset,
             )
 
+        if not self.shared_utils.underlay_multicast and (vlan.vxlan_flood_multicast.enabled or tenant.vxlan_flood_multicast.enabled):
+            msg = "'vxlan_flood_multicast' is only supported in combination with 'underlay_multicast: True'."
+            raise AristaAvdError(msg)
+        if vlan.vxlan_flood_multicast.enabled is not None:
+            vxlan_vlan.flood_group = (
+                vlan.vxlan_flood_multicast.underlay_multicast_group
+                if vlan.vxlan_flood_multicast.underlay_multicast_group and vlan.vxlan_flood_multicast.enabled is True
+                else None
+            )
+        elif tenant.vxlan_flood_multicast.enabled is True:
+            if not tenant.vxlan_flood_multicast.underlay_l2_multicast_group_ipv4_pool:
+                msg = f"'vxlan_flood_multicast.underlay_l2_multicast_group_ipv4_pool' for Tenant: {tenant.name} is required."
+                raise AristaAvdInvalidInputsError(msg)
+            vxlan_vlan.flood_group = self.shared_utils.ip_addressing.evpn_underlay_l2_flood_group(
+                tenant.vxlan_flood_multicast.underlay_l2_multicast_group_ipv4_pool,
```

Comment:
I am not sure this will work well. What happens if I enable under the vlan but wish to rely on the pool to get the group? So not setting enabled to false under the tenant and overriding with true under the vlan.
I think you need to set a vxlan_vlan_flood_multicast_enabled using the default() between the two booleans. And then if the result is true you can look for the direct ip under the vlan or fall back to the pool from the tenant level.